### PR TITLE
feat(Chromecast-Plugin): add a method to get the cast connected device e info

### PR DESCRIPTION
**Pull request by snooky23**
We want to add a new native module to the Chromecast plugin that gives the currently connected device metadata. 
This enables us to create custom Chromecast views in react native.

The current metadata that will be available is:
* deviceID
* deviceVersion
* friendlyName
* modelName

In the future, we would be able to add more params such as volume and more.

### CODE REVIEW

**Review by alexzchut — CHANGES_REQUESTED** [review]
@snooky23 Please move the implementation to the plugin level instead of using GoogleCast on bridge

**Comment by snooky23** [comment]
@alexzchut - done

File: plugins/zapp-generic-chromecast/apple/ios/Bridge/Chromecast/ChromecastManagerBridge.swift [hunk 1]
```diff
@@ -9,6 +9,8 @@
 import Foundation
 import React
 import ZappCore
+import GoogleCast
```
Comment (alexzchut) [inline plugins/zapp-generic-chromecast/apple/ios/Bridge/Chromecast/ChromecastManagerBridge.swift:12]: no need of this

File: plugins/zapp-generic-chromecast/apple/ios/Bridge/Chromecast/ChromecastManagerBridge.swift [hunk 2]
```diff
@@ -115,6 +117,26 @@ class ChromecastManager: NSObject, RCTBridgeModule {
...
+                                   rejecter: @escaping RCTPromiseRejectBlock) {
+        DispatchQueue.main.async {
+            
+            guard let pluginInstance = self.pluginInstance else {
+                rejecter("0", "plugin not available", nil)
+                return
+            }
+            
+            do {
+                let connctedDeviceInfo = try pluginInstance.connctedDeviceInfo()
```
Comment (alexzchut) [inline plugins/zapp-generic-chromecast/apple/ios/Bridge/Chromecast/ChromecastManagerBridge.swift:131]: **connctedDeviceInfo** typo

File: plugins/zapp-generic-chromecast/apple/ios/Extensions/ChromecastAdapter+ChromecastProtocol.swift [hunk 3]
```diff
@@ -64,75 +68,90 @@ extension ChromecastAdapter: ChromecastProtocol {
...
         if !hasConnectedCastSession() {
             return nil
         }
-
+        
         return GCKCastContext.sharedInstance().sessionManager.currentSession?.device.friendlyName
     }
-
+    
+    public func connctedDeviceInfo() throws -> Dictionary<String,Any>? {
```
Comment (alexzchut) [inline plugins/zapp-generic-chromecast/apple/ios/Extensions/ChromecastAdapter+ChromecastProtocol.swift:100]: typo in func name

**Review by alexzchut — CHANGES_REQUESTED** [review]

**Review by alexzchut — APPROVED** [review]

**Review by drazumeev-sc — COMMENTED** [review]
API similliar to Android shared method: https://github.com/applicaster/Zapp-Frameworks/blob/5893a983166b5774040386fc9f01a642e6da9113/plugins/zapp-generic-chromecast/android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java#L222

💯

**Pull request merged**